@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -24,30 +33,40 @@ export class TextElementState {
      * for text part and icon, there is no separate array of [[RenderState]]s for the text parts
      * of the line markers.
      */
-    m_iconRenderStates?: RenderState | RenderState[];
+    private m_iconRenderStates?: RenderState | RenderState[];
 
     /**
      * @hidden
      * Used during rendering.
      */
-    m_textRenderState?: RenderState;
+    private m_textRenderState?: RenderState;
+
+    constructor(
+        private readonly m_textElement: TextElement,
+        groupState: TextElementGroupState,
```

Comment:
Ok, here it will be possible to do what you say, and invalidate the sorting directly from the group state.
But setViewDistance() can be called on any label state from text element renderer and we need to detect that and invalidate the group sorting .